projects/step-frontend/src/lib/modules/execution/services/executions-panels.service.ts

```diff
@@ -34,42 +38,65 @@ export class ExecutionsPanelsService {
     }
     (this as FieldsAccessor).customPanels = this._viewRegistry.getDashlets('execution');
     this.customPanels.forEach(({ id, label }) => {
-      this._panels[id] = { label, show: true, enabled: true };
+      this._defaultPanels[id] = { label, show: true, enabled: true };
     });
+  }
 
-    (this as FieldsAccessor).panels = Object.entries(this._panels).map(([id, { label }]) => ({ id, label }));
+  getPanel(viewId: string, eid: string): ExecutionStepPanel | undefined {
+    if (this._panels[eid]) {
+      return this._panels[eid][viewId];
+    }
+    return this._defaultPanels[viewId];
   }
 
-  getPanel(viewId: string): ExecutionStepPanel | undefined {
-    return this._panels[viewId];
+  observePanel(viewId: string, eid: string): Observable<ExecutionStepPanel | undefined> {
+    if (!this._panels$[eid] || !this._panels$[eid][viewId]) {
```

Comment:
Can be shorted as
`!this._panels$?.[eid]?.[viewId]`

Reply:
Thank you, I was always wondering why we can't do this but did not know about "?."